docs/src/components/ExpoSnack.tsx

```diff
@@ -0,0 +1,153 @@
+import * as React from 'react';
+import { View } from '@aws-amplify/ui-react';
+
+// https://github.com/expo/snack/blob/main/docs/url-query-parameters.md#Files
+interface File {
+  type?: 'CODE' | 'ASSET';
+  contents?: string;
+  url?: string;
+}
+
+// https://github.com/expo/snack/blob/main/docs/url-query-parameters.md#parameters
+interface SnackOptions {
+  name?: string;
+  description?: string;
+  code?: string;
+  platform?: 'web' | 'android' | 'ios';
+  supportedPlatforms?: string;
+  theme?: 'light' | 'dark';
+  preview?: boolean;
+  loading?: string;
+  dependencies?: string[];
+  files?: Record<string, File>;
+  sdkVersion?: string;
+}
+
+const defaultOptions = {
+  name: 'example',
+  platform: 'android',
+  supportedPlatforms: 'ios,android',
+  preview: true,
+  loading: 'lazy',
+  sdkVersion: '45.0.0',
+  dependencies: [
+    '@aws-amplify/ui-react-native,aws-amplify',
+    'react-native-safe-area-context',
+    'amazon-cognito-identity-js',
+    '@react-native-community/netinfo',
+    '@react-native-async-storage/async-storage',
+    'react-native-get-random-values',
+    'react-native-url-polyfill',
+  ],
+};
+
+const exportsCode = `// NOTE: this file needs to be generated by the Amplify CLI
+export default {}`;
+
+export const ExpoSnack = (options: SnackOptions) => {
+  const {
+    code,
+    files,
+    dependencies,
+    name,
+    description,
+    preview,
+    platform,
+    supportedPlatforms,
+    sdkVersion,
+    ...rest
+  } = {
+    ...defaultOptions,
+    ...options,
+  };
+  // We need a unique ID b/c we need to listen to window events that the iframe
+  // will send and it sends the iframe id
+  const id = Math.random().toString(36).substring(2, 10);
```

Comment:
Does this value need a ref to prevent a new value from being generated on rerender?